Sofa/framework/Core/src/sofa/core/STLParallelismSupport.h

```diff
@@ -0,0 +1,50 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+// execution policies only supported by MSVC >=2019 and GCC >=10
+#ifdef _MSC_VER
```

Comment:
Isn't the use case for the macro ?
 __cpp_lib_parallel_algorithm 

https://en.cppreference.com/w/cpp/feature_test
https://en.cppreference.com/w/cpp/algorithm#Execution_policies

Reply:
I did not the existence of these macros 👍
But seems supported from c++20 no ? (` Feature testing (since C++20)` )

Reply:
From what I understand from https://en.cppreference.com/w/cpp/feature_test, the macro `__cpp_lib_parallel_algorithm` gives only the year and month when the corresponding feature has been included in the working draft. But it could not be used with `__has_cpp_attribute`.

Reply:
From the example in cppreferences I would have expected something like:
```cpp
#if __cpp_lib_parallel_algorithm >= 201603L               // Check for a specific version of a feature
#  define HAS_SUPPORT_STL_PARALLELISM  1
#else
#  define HAS_SUPPORT_STL_PARALLELISM  0
#endif
```
would do the work.